Depend only on aws-sdk-core

alanraison opened the pull request:
The aws-sdk dependency pulls in a large number of otherwise unnecessary AWS SDKs. Restrict this module's runtime dependency to just aws-sdk-core.

k1LoW commented on .rubocop.yml:
```diff
@@ -1,7 +1,13 @@
 AllCops:
-  TargetRubyVersion: 2.1
```
🙏 https://github.com/k1LoW/awsecrets/pull/24#issuecomment-568826643

alanraison replied:
I wasn't able to build this change with Ruby 2.1 (on Travis) :(

alanraison committed:
Depend only on aws-sdk-core

The aws-sdk dependency pulls in a large number of otherwise unnecessary
AWS SDKs. Restrict this module's runtime dependency to just
aws-sdk-core.

k1LoW commented on awsecrets.gemspec:
```diff
@@ -18,7 +18,7 @@ Gem::Specification.new do |spec|
   spec.require_paths = ['lib']
 
-  spec.add_runtime_dependency 'aws-sdk', '>= 2', '< 4'
+  spec.add_runtime_dependency 'aws-sdk-core', '>= 3', '< 4'
```
awsecrets still support AWS SDK for Ruby V2 🙏
